src/hooks/usePrioritiseTokensForPrice.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from 'react'
+
+import useSafeState from './useSafeState'
+import { TokenDetails } from 'types'
+
+interface HookReturn {
+  baseToken: TokenDetails
+  quoteToken: TokenDetails
+  wasPriorityAdjusted: boolean
+}
+
+interface HookProps {
+  sellToken: TokenDetails
+  receiveToken: TokenDetails
+}
+
+interface SmartTokenStruct {
+  baseToken: TokenDetails
+  quoteToken: TokenDetails
+  wasPriorityAdjusted: boolean
+}
+
+export function checkMarketAndSmartAdjust({ sellToken, receiveToken }: HookProps): SmartTokenStruct {
+  let baseToken = receiveToken
+  let quoteToken = sellToken
+  let wasPriorityAdjusted = false
+
+  // Prio: 1    - STABLE USD
+  // Prio: 2    - STABLE EUR
+  // Prio: 3    - WETH
+  // Prio: 99   - VOLATILE
+  const basePrio = baseToken.priority || 99
```

Comment:
99 is arbitrary, I would use `Number.MAX_SAFE_INTEGER`

Reply:
Also, if you do a small function getPriority, you can refactor it
```js
const basePrio = getPriority(baseToken)
const quotePrio = getPriority(quotePrio)
```

Also, I wouldn't shorten `priority` word

Reply:
You resolved my comment but I still see the same issue...

Reply:
it's in a different section, `useTokenList` but i will remove this as it's no longer needed